components/org.wso2.carbon.identity.api.server.claim.management/org.wso2.carbon.identity.rest.api.server.claim.management.v1/src/main/java/org/wso2/carbon/identity/rest/api/server/claim/management/v1/core/ServerClaimManagementService.java

```diff
@@ -1444,4 +1455,27 @@ private void validateAttributeMappings(List<AttributeMappingDTO> attributeMappin
             }
         }
     }
+
+    private void validateRootOrganization() throws ClaimMetadataClientException {
+
+        String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        try {
+            String organizationId = getOrganizationManager().resolveOrganizationId(tenantDomain);
```

Comment:
We can use PrivilegedCarbonContext.getThreadLocalCarbonContext().getOrganizationId()

Reply:
In case the API is accessed via /t/ path, `PrivilegedCarbonContext.getThreadLocalCarbonContext().getOrganizationId()` will not set. 
So if org id not found need to resolve through tenant domain again

Reply:
Yes, ideally sub-orgs resources are meant to be accessed with /o/ paths. In case it was accessed with /t/ path, the mentioned concern is there. But we can first fetch orgId from carbon context, if not available resolve from tenant-domain.

Anyways, with future improvements for sub-organization resources will be accessed with /t/<primary-tenant> with organization bound token. In such case, organizationId will be populated in the context and won't be null.